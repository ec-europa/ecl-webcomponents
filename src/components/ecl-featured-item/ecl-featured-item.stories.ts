import { randomizedLink } from "../../utils/randomizedLink"; 

const getArgs = () => {
  return {
    variant: 'extended',
    position: 'left',
    itemTitle: 'Title',
    image: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg',
    description: `
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
      Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. 
      Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
    </p>
    <ecl-spacing direction="t" value="s">
      <ecl-link
        variant="standalone"
        style-class="ecl-button sc-ecl-button-ec ecl-button--call"
        href="${randomizedLink('/example.html')}"
      >
        Read more
        <ecl-icon slot="icon-after" icon="corner-arrow" transform="rotate-90" />
      </ecl-link>
    </ecl-spacing>
    `,
    mediaCaption: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['default', 'extended'],
      description: "Featured item variant"
    },
    position: {
      type: { name: 'select'},
      options: ['left', 'right'],
      description: 'Position of the image',
    },
    itemTitle: {
      type: { name: 'string'},
      description: 'Title of the featured item',
    },
    image: {
      type: { name: 'string'},
      description: 'Image for the media container'
    },
    description: {
      type: { name: 'string'},
      description: 'Content of the ecl-featured-item element',
    },
    mediaCaption: {
      type: { name: 'string' },
      description: 'Caption for the media container'
    }
  };
};

export default {
  title: 'Components/featured-item',
};

const Template = (args) => 
`<ecl-featured-item
  variant="${args.variant}"
  theme="${args.theme}"
  item-title="${args.itemTitle}"
  position="${args.position}"
  image="${args.image}"
  media-caption="${args.mediaCaption}"
>
  ${args.description}
</ecl-featured-item>`;

export const FeaturedItem = Template.bind({});
FeaturedItem.storyName = 'default';
FeaturedItem.args = getArgs();
FeaturedItem.argTypes = getArgTypes();
