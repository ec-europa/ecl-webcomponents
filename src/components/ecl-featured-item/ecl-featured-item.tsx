import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-featured-item',
  styleUrls: {
    ec: './build/styles/ecl-featured-item-ec.css',
    eu: './build/styles/ecl-featured-item-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclFeaturedItem {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() itemTitle: string;
  @Prop() image: string;
  @Prop() mediaCaption: string;
  @Prop() position: string = 'left';
  @Prop() eclScript: boolean = false;
  @Prop() defaultContainerClass = 'ecl-featured-item__container';

  getClass(): string {
    return [
      `ecl-featured-item`,
      `ecl-featured-item--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    const containerClasses = this.position === 'right' ? `${this.defaultContainerClass} ecl-featured-item__container--right` : this.defaultContainerClass;

    return (
      <article
        class={this.getClass()}
      >
        <div class={containerClasses}>
          <div class="ecl-featured-item__item">
            <ecl-media-container
              styleClass="ecl-featured-item__media_container"
              image={this.image}
              hasCaption={this.mediaCaption !== ''}
            >
             {this.mediaCaption}
            </ecl-media-container>
          </div>
          <div class="ecl-featured-item__item">
          { this.itemTitle ?
            <div class="ecl-featured-item__title">
              {this.itemTitle}
            </div> : '' }
            <div class="ecl-featured-item__description">
              <slot></slot>
            </div>
          </div>
        </div>
      </article>
    );
  }
}
