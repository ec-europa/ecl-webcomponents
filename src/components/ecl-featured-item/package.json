{
  "name": "@ecl/web-components-featured-item",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.0.0",
  "description": "ECL Featured item",
  "main": "ecl-featured-item.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.0",
    "@ecl/vanilla-component-featured-item": "3.7.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}