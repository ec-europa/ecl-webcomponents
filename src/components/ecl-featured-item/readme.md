# ecl-featured-item



<!-- Auto Generated Below -->


## Properties

| Property                | Attribute                 | Description | Type      | Default                          |
| ----------------------- | ------------------------- | ----------- | --------- | -------------------------------- |
| `defaultContainerClass` | `default-container-class` |             | `string`  | `'ecl-featured-item__container'` |
| `eclScript`             | `ecl-script`              |             | `boolean` | `false`                          |
| `image`                 | `image`                   |             | `string`  | `undefined`                      |
| `itemTitle`             | `item-title`              |             | `string`  | `undefined`                      |
| `mediaCaption`          | `media-caption`           |             | `string`  | `undefined`                      |
| `position`              | `position`                |             | `string`  | `'left'`                         |
| `styleClass`            | `style-class`             |             | `string`  | `undefined`                      |
| `theme`                 | `theme`                   |             | `string`  | `'ec'`                           |
| `variant`               | `variant`                 |             | `string`  | `'default'`                      |
| `withUtils`             | `with-utils`              |             | `boolean` | `false`                          |


## Dependencies

### Depends on

- [ecl-media-container](../ecl-media-container)

### Graph
```mermaid
graph TD;
  ecl-featured-item --> ecl-media-container
  style ecl-featured-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
