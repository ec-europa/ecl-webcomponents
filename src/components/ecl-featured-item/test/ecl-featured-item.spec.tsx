import { newSpecPage } from '@stencil/core/testing';
import { EclFeaturedItem } from '../ecl-featured-item';

describe('ecl-featured-item', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclFeaturedItem],
      html: `<ecl-featured-item
        variant="extended"
        theme="ec"
        item-title="Title"
        position="left"
        image="https://inno-ecl.s3.amazonaws.com/media/example.htmls/example.html-image.jpg"
        media-caption="Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
      >
         <p class="ecl-u-type-paragraph-m">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.</p>
      </ecl-featured-item>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
