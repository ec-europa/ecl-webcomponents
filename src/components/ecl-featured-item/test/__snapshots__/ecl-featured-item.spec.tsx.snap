// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-featured-item renders 1`] = `
<ecl-featured-item image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" item-title="Title" media-caption="Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur." position="left" theme="ec" variant="extended">
  <!---->
  <article class="ecl-featured-item ecl-featured-item--extended">
    <div class="ecl-featured-item__container">
      <div class="ecl-featured-item__item">
        <ecl-media-container hascaption="" image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" styleclass="ecl-featured-item__media_container">
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
        </ecl-media-container>
      </div>
      <div class="ecl-featured-item__item">
        <div class="ecl-featured-item__title">
          Title
        </div>
        <div class="ecl-featured-item__description">
          <p class="ecl-u-type-paragraph-m">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
          </p>
        </div>
      </div>
    </div>
  </article>
</ecl-featured-item>
`;
